app/src/main/java/com/bogdan/codeforceswatcher/network/UsersRequest.kt

```diff
@@ -1,65 +1,50 @@
 package com.bogdan.codeforceswatcher.network
 
 import com.bogdan.codeforceswatcher.features.users.models.User
-import com.bogdan.codeforceswatcher.features.users.redux.requests.RatingChangeResponse
-import com.bogdan.codeforceswatcher.features.users.redux.requests.UsersResponse
 import com.bogdan.codeforceswatcher.network.models.Error
 import com.bogdan.codeforceswatcher.network.models.UsersRequestResult
 import kotlinx.coroutines.*
-import retrofit2.Call
-import retrofit2.Callback
-import retrofit2.Response
 
 fun getUsers(handles: String, isRatingUpdatesNeeded: Boolean, onCompleted: (UsersRequestResult) -> Unit) {
-    val userCall = RestClient.getUsers(handles)
-    userCall.enqueue(object : Callback<UsersResponse> {
-
-        override fun onResponse(call: Call<UsersResponse>, response: Response<UsersResponse>) {
+    GlobalScope.launch {
```

Comment:
`GlobalScope` isn't good for production apps as its lifecycle is not bound to anything. Use `CoroutineScope(Dispatchers.Main)` instead and the you don't need to switch there when needed. Here is an example:

```
CoroutineScope(Dispatchers.Main).launch {
        try {
            val response = RestClient.getUsers(handles)
            response.body()?.users?.let { users ->
                if (isRatingUpdatesNeeded) {
                    onCompleted(loadRatingUpdates(users))
                } else {
                    onCompleted(UsersRequestResult.Success(users))
                }
            } ?: onCompleted(UsersRequestResult.Failure(Error.RESPONSE))
        } catch (t: Throwable) {
            onCompleted(UsersRequestResult.Failure(Error.INTERNET))
        }
    }
```

But you need to move `getRating` call to `coroutines` too, of course.